### PR TITLE
vlc: update to 3.0.7.1

### PR DESCRIPTION
**Pull request by michaellass**
#### Description
Update VLC to version 3.0.7.1. This updates contains several security patches. See https://www.videolan.org/developers/vlc-branch/NEWS.

Two patches can be dropped as they are included upstream by now.

I only tested VLC and not libVLC. However I would not expect any breakage since the packaging is mostly untouched.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.5 18F132
Xcode 10.2.1 10E1001

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`? (--> no new errors)
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`? (--> only without trace mode)
- [x] tested basic functionality of all binary files? (only of VLC, not libVLC)

### CODE REVIEW

File: multimedia/VLC/Portfile [hunk 1]
```diff
@@ -50,7 +50,7 @@ if {${subport} eq ${name}} {
 universal_variant   no
 supported_archs     x86_64
 
-version             3.0.6
+version             3.0.7.1
 revision            1
```
Comment (chrstphrchvz) [inline multimedia/VLC/Portfile]: Please reset the revision to 0

Comment (michaellass) [reply multimedia/VLC/Portfile]: Oops, good catch. Thanks!

**Comment by michaellass** [comment]
All the Travis builds just timed out. However, I'm not sure why the Azure build for macOS 10.13 failed and if it is a new problem. Here's the first error message:

```
access/avcapture.m:50:43: warning: declaration of 'struct AVMediaType' will not be visible outside of this function [-Wvisibility]
+ (void)requestAccessForMediaType:(struct AVMediaType)mediaType completionHandler:(void (^)(BOOL granted))handler API_AVAILABLE(macos(10.14), ios(7.0));
                                          ^
access/avcapture.m:345:57: error: argument type 'struct AVMediaType' is incomplete
            [AVCaptureDevice requestAccessForMediaType: AVMediaTypeVideo completionHandler:^(BOOL granted) {
                                                        ^~~~~~~~~~~~~~~~
access/avcapture.m:50:43: note: forward declaration of 'struct AVMediaType'
+ (void)requestAccessForMediaType:(struct AVMediaType)mediaType completionHandler:(void (^)(BOOL granted))handler API_AVAILABLE(macos(10.14), ios(7.0));
                                          ^
1 warning and 1 error generated.
make[4]: *** [access/libavcapture_plugin_la-avcapture.lo] Error 1
```
Full log: https://paste.z0k.xyz/fc427204b132

Has anyone interested in this package 10.13 at hand to test? I unfortunately do not.

**Comment by RJVB** [comment]
`VLC.xcodeproj` is now called `vlc.xcodeproj` so the reinplace commands need to be updated (line 430 or so).

Other than that the libVLC subport builds fine on my 10.9.5 system (`+dbus+qtkit+quartz+shout+smb+svg`), so LGTM.

**Comment by michaellass** [comment]
> `VLC.xcodeproj` is now called `vlc.xcodeproj` so the reinplace commands need to be updated (line 430 or so).

That's fixed now as well. Thanks for testing libVLC. Do you have any idea about the Azure buildbot failure on 10.13?

**Comment by RJVB** [comment]
>That's fixed now as well. Thanks for testing libVLC. Do you have any idea about the Azure buildbot failure on 10.13?

No, not off the top of my hat... I don't even know what Azure is (a quick search in other ports suggests is some kind of cloud platform?)

**Comment by chrstphrchvz** [comment]
> I don't even know what Azure is (a quick search in other ports suggests is some kind of cloud platform?)

Azure Pipelines CI is an alternative to Travis, as I understand it. The main differences I'm aware of are longer job runtime limit (6 hours) and what macOS versions are available (10.13 and 10.14 only). It's only been around less than a year.

**Comment by RJVB** [comment]
Well, any questions about that failing would have to be put to the people who set it up for MacPorts, they should be able to understand the reason for the error (I'd hope ;))

**Comment by l2dy** [comment]
@RJVB @chrstphrchvz Did you test this update on macOS 10.13? If it builds successfully on macOS 10.13, we can merge this regardless of CI errors.

**Comment by RJVB** [comment]
>@RJVB @chrstphrchvz Did you test this update on macOS 10.13? If it builds successfully on macOS 10.13, we can merge this regardless of CI errors.

Not I (can't) but I see no reason why it'd fail on 10.13 if it builds correctly on the supported earlier OS versions (does it?)

**Comment by chrstphrchvz** [comment]
I have not tested this update nor do I have access to 10.13 currently.

**Comment by michaellass** [comment]
I just now see that one of our patches touches the code that's causing the failure. I just pushed a commit to this PR which removes this patch, mainly to let the buildbot figure out if it is actually causing the issue. If it is, we need to figure out if and on what macOS versions that patch is actually required.

**Comment by chrstphrchvz** [comment]
Both Azure builds have finished successfully.

**Comment by michaellass** [comment]
@RJVB could you test if the build still works for you on macOS 10.9 and we can just drop that patch?

**Comment by RJVB** [comment]
I am 99.99% certain it still won't build without that patch on 10.9 . Would you mind bisecting which hunk causes the problem? My money would be on 

```
--- modules/access/orig.avcapture.m	2018-10-28 12:11:19.000000000 +0100
+++ modules/access/avcapture.m	2019-02-18 00:27:36.000000000 +0100
@@ -43,13 +43,15 @@
 #import <AVFoundation/AVFoundation.h>
 #import <CoreMedia/CoreMedia.h>
 
+#ifdef MAC_OS_X_VERSION_10_13
 #ifndef MAC_OS_X_VERSION_10_14
 @interface AVCaptureDevice (AVCaptureDeviceAuthorizationSince10_14)
 
-+ (void)requestAccessForMediaType:(AVMediaType)mediaType completionHandler:(void (^)(BOOL granted))handler API_AVAILABLE(macos(10.14), ios(7.0));
++ (void)requestAccessForMediaType:(struct AVMediaType)mediaType completionHandler:(void (^)(BOOL granted))handler API_AVAILABLE(macos(10.14), ios(7.0));
 
 @end
 #endif
+#endif
 
 /*****************************************************************************
 * Local prototypes
```


Were there failures on 10.12 and/or 10.11, and for what OS version are we certain that the build succeeds WITHOUT the patch?
The easiest workaround would be to apply the patch only on the untested older systems and then correct if necessary.

**Comment by RJVB** [comment]
News flash: according to my own repo this patch was added as a fix for trac #58206, to fix building on OS X 10.11 . So the easy workaround would be to apply the patch only for `${os.major} <= 15` .

>>However, I'm not sure why the Azure build for macOS 10.13 failed and if it is a new problem. Here's the first error message:

It is a strange one indeed:

```
>access/avcapture.m:50:43: warning: declaration of 'struct AVMediaType' will not be visible outside of this function [-Wvisibility]
```

That one is probably indicative: this means that the AVMediaType type hasn't been declared, DESPITE the #import AVFoundation.h statement above.

So this one is expectable:
```
>+ (void)requestAccessForMediaType:(struct AVMediaType)mediaType completionHandler:(void (^)(BOOL granted))handler API_AVAILABLE(macos(10.14), ios(7.0));
>                                          ^
>access/avcapture.m:345:57: error: argument type 'struct AVMediaType' is incomplete
>            [AVCaptureDevice requestAccessForMediaType: AVMediaTypeVideo completionHandler:^(BOOL granted) {
>                                                        ^~~~~~~~~~~~~~~~
>access/avcapture.m:50:43: note: forward declaration of 'struct AVMediaType'
```

Yet, the documentation (https://developer.apple.com/documentation/avfoundation/avmediatype) suggests that there is no need to include anything else for this type, which exists on 10.13 and up.

This part is weird too btw, notice the exclusion of 10.14 combined with the "since 10.14" hint in the protocol name:

```
 #ifndef MAC_OS_X_VERSION_10_14
 @interface AVCaptureDevice (AVCaptureDeviceAuthorizationSince10_14)
```

In short, I don't understand why this builds withOUT the patch if it doesn't build WITH the patch.

What's the exact 10.13 version this was running on? VideoLan suggest using at least 10.13.4 for building VLC *4* ...

Would you mind testing with this hacked version of the patch (which doesn't introduce the exact change that is erroring out)?

```
diff --git modules/access/qtsound.m modules/access/qtsound.m
index af190d1..3f0fc4f 100644
--- modules/access/orig.qtsound.m	2018-10-28 12:11:19.000000000 +0100
+++ modules/access/qtsound.m	2019-02-18 00:41:40.000000000 +0100
@@ -45,6 +45,7 @@
 #import <QTKit/QTKit.h>
 #import <AVFoundation/AVFoundation.h>
 
+#ifdef MAC_OS_X_VERSION_10_13
 #ifndef MAC_OS_X_VERSION_10_14
 @interface AVCaptureDevice (AVCaptureDeviceAuthorizationSince10_14)
 
@@ -52,6 +53,7 @@
 
 @end
 #endif
+#endif
 
 /*****************************************************************************
  * Local prototypes.
@@ -318,6 +320,7 @@
             goto error;
         }
 
+#ifdef MAC_OS_X_VERSION_10_13
         if (@available(macOS 10.14, *)) {
             msg_Dbg(p_demux, "Check user consent for access to the audio device");
 
@@ -334,6 +337,7 @@
                 goto error;
             }
         }
+#endif
 
         audioInput = [[QTCaptureDeviceInput alloc] initWithDevice: p_sys->audiodevice];
         if(!audioInput) {
diff --get modules/access/avcapture.m modules/access/avcapture.m
index af190d1..3f0fc4f 100644
--- modules/access/orig.avcapture.m	2018-10-28 12:11:19.000000000 +0100
+++ modules/access/avcapture.m	2019-02-18 00:27:36.000000000 +0100
@@ -43,13 +43,15 @@
 #import <AVFoundation/AVFoundation.h>
 #import <CoreMedia/CoreMedia.h>
 
+#ifdef MAC_OS_X_VERSION_10_13
 #ifndef MAC_OS_X_VERSION_10_14
 @interface AVCaptureDevice (AVCaptureDeviceAuthorizationSince10_14)
 
-+ (void)requestAccessForMediaType:(AVMediaType)mediaType completionHandler:(void (^)(BOOL granted))handler API_AVAILABLE(macos(10.14), ios(7.0));
++ (void)requestAccessForMediaType:(AVMediaType)mediaType completionHandler:(void (^)(BOOL granted))handler API_AVAILABLE(macos(10.14), ios(7.0));
 
 @end
 #endif
+#endif
 
 /*****************************************************************************
 * Local prototypes
@@ -334,6 +336,7 @@
             goto error;
         }
 
+#ifdef MAC_OS_X_VERSION_10_13
         if (@available(macOS 10.14, *)) {
             msg_Dbg(p_demux, "Check user consent for access to the video device");
 
@@ -350,6 +353,7 @@
                 goto error;
             }
         }
+#endif
 
         input = [AVCaptureDeviceInput deviceInputWithDevice:(__bridge AVCaptureDevice *)p_sys->device error:&o_returnedError];
 

```

**Comment by l2dy** [comment]
> What's the exact 10.13 version this was running on? VideoLan suggest using at least 10.13.4 for building VLC *4* ...

10.13.6, according to https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/macos/macos-10.13-Readme.md.

**Comment by l2dy** [comment]
@RJVB FYI, to use Markdown in your comment, you need to post your comment on the GitHub web interface. Comments submitted via email reply won't be rendered in Markdown.

**Comment by michaellass** [comment]
@RJVB The formatting of your comment didn't really work out. But I assume you just want to leave the one line that causes the underlying warning untouched. I just pushed another version here, so let's see what the buildbots think about it.

Regarding the exact version of the 10.13 buildbot: The build log mentions "darwin/17.7.0" which also matches macOS 10.13.6.

**Comment by RJVB** [comment]
> Comments submitted via email reply won't be rendered in Markdown.

Weird, and highly annonying. I'm pretty certain that used to work with other projects...

**Comment by michaellass** [comment]
Both builds passed, so the simplified patch does not break 10.13. Shall we stick to that solution for now? Maybe you can retest compilation on 10.9 before merging.

**Comment by RJVB** [comment]
Does anyone have access to 10.11 for testing? I don't have access to my Mac right now but the fact someone had to file a ticket suggests (very strongly, in fact) that this code doesn't get compiled on 10.9 - otherwise I would have run into similar issues as reported on that trac ticket.

So I'm a bit in dubio. Either we commit the patch this way and hope the inline declaration of AVMediaType isn't required (and improvise if we get reports for failures on 10.11) OR we simply apply the patch only on 10.11 and below. I would probably do that if I had commit powers (and no access to 10.11).
`accapture.m` hasn't changed since 3.0.6 so the patched file shouldn't all of a sudden fail to build on 10.11 (if there are no relevant changes in the headerfiles it includes either).

**Comment by l2dy** [comment]
VLC 3.0.7 is a security release, so merging now and we can fix failed builds later.

**Comment by l2dy** [comment]
Build failed on 10.11. Log: https://build.macports.org/builders/ports-10.11_x86_64-builder/builds/88636/steps/install-port/logs/stdio.

```
video_output/macosx.m:342:13: error: unexpected '@' in program
        if (@available(macOS 10.14, *)) {
            ^
video_output/macosx.m:497:9: warning: 'NSOpenGLPFAWindow' is deprecated: first deprecated in macOS 10.9 [-Wdeprecated-declarations]
        NSOpenGLPFAWindow,
        ^
/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGL.h:86:2: note: 'NSOpenGLPFAWindow' has been explicitly marked deprecated here
        NSOpenGLPFAWindow NS_ENUM_DEPRECATED_MAC(10_0, 10_9)           =  80,
        ^
1 warning and 1 error generated.
mv -f video_output/opengl/.deps/libgl_plugin_la-display.Tpo video_output/opengl/.deps/libgl_plugin_la-display.Plo
make[4]: *** [video_output/macosx.lo] Error 1
```

**Comment by michaellass** [comment]
That code part was introduced in 3.0.7: https://github.com/videolan/vlc-3.0/commit/a8539f7e41e9086ff6a1a15c323044ef18031d9e

We can probably just revert that for macOS < 10.13. However, I fear that it will become increasingly difficult to make new VLC versions build on older versions of macOS.

**Comment by RJVB** [comment]
>Build failed on 10.11. Log: https://build.macports.org/builders/ports-10.11_x86_64-builder/builds/88636/steps/install-port/logs/stdio.
>
>```
>video_output/macosx.m:342:13: error: unexpected '@' in program
>        if (@available(macOS 10.14, *)) {
>            ^

So no build error on 10.14 with the pre1014 patch after all?

Either way it seems some version of that patch will remain necessary on 10.11 (and below), and will require a hunk to exclude this `@available()` statement. 

> However, I fear that it will become increasingly difficult to make new VLC
> versions build on older versions

That much is certain, let's just hope that it remains possible for the remaining 3.x series so that we can pull a similar trick with a port:VLC3 as we did with VLC2...

**Comment by michaellass** [comment]
@RJVB That's another issue now, not related to that pre1014 patch. If I see this correctly, the build currently fails for 10.9, 10.10 and 10.11.

https://build.macports.org/builders/ports-10.11_x86_64-watcher/builds/25462
https://build.macports.org/builders/ports-10.10_x86_64-watcher/builds/25508
https://build.macports.org/builders/ports-10.11_x86_64-watcher/builds/25462

Reverting a8539f7 on those versions should fit it. I'll create another PR for that.

**Comment by RJVB** [comment]
>@RJVB That's another issue now, not related to that pre1014 patch. If I see this correctly, the build currently fails for 10.9, 10.10 and 10.11.

The simplified patch makes the build fail on 10.9? Strange... guess I'm going to have to try again this afternoon then.

What I meant is that the fix for this issue is something that should be added to the pre1014 patch because that's where it belongs.

**Comment by michaellass** [comment]
I don't see any connection between the pre-1014 patch we needed to adjust and the new problem discovered by the macports builders. macosx.m is not touched by any of the existing patches. Do I miss something here?

I didn't look into details of the failed builds. Maybe 10.9 is a false alert because as far as I understand VLC does not build there anyway and only libVLC does.

**Comment by RJVB** [comment]
> I don't see any connection between the pre-1014 patch we needed to adjust and the new problem discovered by the macports builders. macosx.m is not touched by any of the existing patches. Do I miss something here?

There is no connection with the adjustment of the pre1014 patch but that patch is there to address building issues on systems running OS X < 10.14 and already includes a protection against an `@available()` statement. I like to organise my patches by function/theme, not by target file - at least as long as they're not addressing an isolated issue that is likely to be addressed by a bug fix. That's not the case here.

**Comment by michaellass** [comment]
> There is no connection with the adjustment of the pre1014 patch but that patch is there to address building issues on systems running OS X < 10.14 and already includes a protection against an `@available()` statement. I like to organise my patches by function/theme, not by target file - at least as long as they're not addressing an isolated issue that is likely to be addressed by a bug fix. That's not the case here.

Ah, now I see what you mean. I closed my PR, so you can implement your proposed solution.

**Comment by RJVB** [comment]
Thx.

Maybe not this afternoon though .... powercut :(

**Pull request merged**